samples/snippets/src/test/java/com/example/bigtable/BigtableBaseTest.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigtable;
+
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.models.CreateTableRequest;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.models.BulkMutation;
+import com.google.cloud.bigtable.data.v2.models.Mutation;
+import com.google.protobuf.ByteString;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Random;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.Before;
+
+public class BigtableBaseTest {
+
+  // Environment variables
+  public static String projectId;
+  public static String instanceId;
+
+  // Test output stream
+  private static PrintStream originalOut;
+  public ByteArrayOutputStream bout;
+
+  // Mobile time series db variables
+  public static final String TABLE_ID = generateTableId("mobile-time-series");
```

Comment:
I think this class could be broken up a bit. The generic stuff that happens in each test (`@Before` setup, `@After` teardown, etc) should remain here. However, the mobile time series variables and methods are not relevant to every sample and are a bit confusing to have available in things like the channel pooling example. I would take all of that and put it into a static helper class to be called by those tests instead.

Reply:
done